[DDW-615] Move the GPU Safe Mode under "Help and Support" submenu and add a confirmation dialog

DeeJayElly opened the pull request:
This PR creates new submenu "Help and Support" with new items: "GPU Safe Mode", "Download Logs", "Support Request", and "Known Issues".

## Todos

- [x] Add confirmation dialog for GPU Safe Mode
- [x] Add jp translations for submenu items

## Screenshots
<img width="471" alt="Screenshot 2019-04-10 09 57 04" src="https://user-images.githubusercontent.com/15862062/55962804-e5ebb380-5c71-11e9-8c36-bc83516d51b8.png">
<img width="1148" alt="Screenshot 2019-04-11 15 47 01" src="https://user-images.githubusercontent.com/15862062/55962727-c5bbf480-5c71-11e9-8522-5ddddd76f3d3.png">
<img width="1147" alt="Screenshot 2019-04-11 15 46 46" src="https://user-images.githubusercontent.com/15862062/55962730-c5bbf480-5c71-11e9-800d-8f00326e7ae0.png">
<img width="418" alt="Screenshot 2019-04-11 15 46 27" src="https://user-images.githubusercontent.com/15862062/55962732-c6548b00-5c71-11e9-81f8-32c5981ad306.png">
<img width="417" alt="Screenshot 2019-04-11 15 46 12" src="https://user-images.githubusercontent.com/15862062/55962733-c6548b00-5c71-11e9-838a-e2820d965934.png">

<img width="1151" alt="Screenshot 2019-05-14 22 04 15" src="https://user-images.githubusercontent.com/15862062/57728692-cf7fb000-7694-11e9-9d6f-a09ba841c8c1.png">

## Testing Checklist

- [Slack QA thread](https://input-output-rnd.slack.com/archives/GGKFXSKC6/p1554991463114400)

## Translations Checklist

- Japanese translations (https://docs.google.com/document/d/1pSXejIbW5DjLAO-rgL6zWr990ciVYXvkDTLSnXfc-xU/edit)

---

## Review Checklist

### Basics

- [ ] PR has been assigned and has appropriate labels (`feature`/`bug`/`chore`, `release-x.x.x`)
- [ ] PR is updated to the most recent version of the target branch (and there are no conflicts)
- [ ] PR has a good description that summarizes all changes and shows some screenshots or animated GIFs of important UI changes
- [ ] CHANGELOG entry has been added to the top of the appropriate section (*Features*, *Fixes*, *Chores*) and is linked to the correct PR on GitHub
- [ ] Automated tests: All acceptance and unit tests are passing (`yarn test`)
- [ ] Manual tests (minimum tests should cover newly added feature/fix): App works correctly in *development* build (`yarn dev`)
- [ ] Manual tests (minimum tests should cover newly added feature/fix): App works correctly in *production* build (`yarn package` / CI builds)
- [ ] There are no *flow* errors or warnings (`yarn flow:test`)
- [ ] There are no *lint* errors or warnings (`yarn lint`)
- [ ] There are no *prettier* errors or warnings (`yarn prettier:check`)
- [ ] There are no missing translations (running `yarn manage:translations` produces no changes)
- [ ] Text changes are proofread and approved (Jane Wild / Amy Reeve)
- [ ] Japanese text changes are proofread and approved (Junko Oda)
- [ ] UI changes look good in all themes (Alexander Rukin)
- [ ] Storybook works and no stories are broken (`yarn storybook`)
- [ ] In case of dependency changes `yarn.lock` file is updated

### Code Quality
- [ ] Important parts of the code are properly commented and documented
- [ ] Code is properly typed with flow
- [ ] React components are split-up enough to avoid unnecessary re-renderings
- [ ] Any code that only works in main process is neatly separated from components

### Testing
- [ ] New feature/change is covered by acceptance tests
- [ ] New feature/change is manually tested and approved by QA team
- [ ] All existing acceptance tests are still up-to-date
- [ ] New feature/change is covered by Daedalus Testing scenario
- [ ] All existing Daedalus Testing scenarios are still up-to-date

### After Review
- [ ] Merge the PR
- [ ] Delete the source branch
- [ ] Move the ticket to `done` column on the YouTrack board
- [ ] Update Slack QA thread by marking it with a green checkmark

nikolaglumac commented:
@DeeJayElly also the GPU dialog on my Mac looks like this:
![Screen Shot 2019-04-29 at 12 34 34](https://user-images.githubusercontent.com/376611/56890968-3ab56980-6a7b-11e9-8aa7-cbd07dfda5ae.png)
... there is no "No" option - I have "Cancel" instead. Not sure if this can be altered?

darko-mijic commented:
First parts of the copy are missing.

![image](https://user-images.githubusercontent.com/4280521/56897288-65f58400-6a8e-11e9-9aee-2f37311ff269.png)

DeeJayElly commented:
> @DeeJayElly also the GPU dialog on my Mac looks like this:
> ![Screen Shot 2019-04-29 at 12 34 34](https://user-images.githubusercontent.com/376611/56890968-3ab56980-6a7b-11e9-8aa7-cbd07dfda5ae.png)
> ... there is no "No" option - I have "Cancel" instead. Not sure if this can be altered?

Yes i can change that

DeeJayElly commented:
> First parts of the copy are missing.
> 
> ![image](https://user-images.githubusercontent.com/4280521/56897288-65f58400-6a8e-11e9-9aee-2f37311ff269.png)

That part is in the dialog title and probably not visible on mac osx, while visible on windows, because those are the system dialogs.

DeeJayElly commented:
> > @DeeJayElly also the GPU dialog on my Mac looks like this:
> > ![Screen Shot 2019-04-29 at 12 34 34](https://user-images.githubusercontent.com/376611/56890968-3ab56980-6a7b-11e9-8aa7-cbd07dfda5ae.png)
> > ... there is no "No" option - I have "Cancel" instead. Not sure if this can be altered?
> 
> Yes i can change that

Done

DeeJayElly commented:
> > First parts of the copy are missing.
> > ![image](https://user-images.githubusercontent.com/4280521/56897288-65f58400-6a8e-11e9-9aee-2f37311ff269.png)
> 
> That part is in the dialog title and probably not visible on mac osx, while visible on windows, because those are the system dialogs.

Added @darko-mijic

DeeJayElly commented:
> @DeeJayElly nice work!
> The only thing left to be done is to implement functionality for "Download Logs", "Known Issues" and "Support Request" menu items.

Ok do we have defined functionalities for all of those somewhere, or we are leaving this for seperate task.

nikolaglumac commented:
@DeeJayElly we have all of it implemented but in other parts of the app. Perhaps @daniloprates can assist you.

nikolaglumac commented:
It should all be done in the scope of this task.

DeeJayElly commented:
@nikolaglumac 

Added Notifications implementation to App component and refactored notifications for Loading and Settings component.

Everything works now.

nikolaglumac commented on source/renderer/app/App.js:
```diff
@@ -14,25 +14,57 @@ import type { ActionsMap } from './actions/index';
     const mobxDevTools = global.environment.mobxDevTools ? <DevTools /> : null;
     const { currentTheme } = stores.profile;
     const themeVars = require(`./themes/daedalus/${currentTheme}.js`).default;
+    const notificationOrder = 99999;
```
@DeeJayElly please add a comment regarding this order 🙏

DeeJayElly replied:
@nikolaglumac done

nikolaglumac commented on source/main/utils/buildAppMenus.js:
```diff
@@ -49,8 +55,34 @@ export const buildAppMenus = async (
+
+  const networkLocale = getLocale(network);
+
+  const supportRequestData = {
```
@DeeJayElly please see if we can use a generic solution where we define the support request payload and url in common and use both in main and rendering process 🙏

nikolaglumac replied:
@DeeJayElly any news on this one?

DeeJayElly replied:
Didn't found any suitable solution. I have a feeling that we should leave it like this for now, if you agree @nikolaglumac

nikolaglumac replied:
OK. I have added a card for that so that we improve it in the future: https://iohk.myjetbrains.com/youtrack/issue/DDW-689

DeeJayElly commented:
> @DeeJayElly Please add a quotation mark when the GPU mode is enabled.
> ![image](https://user-images.githubusercontent.com/45355947/58091272-d9903a00-7bd1-11e9-9ba4-49ea9aa94ad4.png)

Fixed

DeeJayElly commented:
> @DeeJayElly @nikolaglumac Hello Aleksandar! Could you please adjust the Defaulted option for the pop-up message after GPU Safe Mode is selected. Currently default options are 1.Windows - Yes 2.Linux - Cancel. For all the OSes i suggest to have "Cancel" as default one. Thank you!

Fixed. Windows was the problem, Mac and Linux work as expected.

DeeJayElly commented:
> @DeeJayElly @nikolaglumac Please review suggested changes.
> 
> 1. I suggest to disable the "Download Logs" link in the Settings Screen if same process was requested from the ToolBar. We have this implemented for Setting screen already. The link is activated only when logs are downloaded.
> 2. The Bug - If you Press "Download Logs" from the Settings screen and then select Cancel the Download link becomes inactive until you switch between the pages.
>    Attaching record for details about both questions.

Done
